source/ranges.tex

```diff
@@ -321,16 +321,11 @@
   \libconcept{input_or_output_iterator}.
 
 \item
-  Otherwise, \tcode{\placeholdernc{decay-copy}(begin(E))} if it is a
-  valid expression and its type \tcode{I} models
-  \libconcept{input_or_output_iterator} with overload
-  resolution performed in a context that includes the declarations:
-  \begin{codeblock}
-  template<class T> void begin(T&&) = delete;
-  template<class T> void begin(initializer_list<T>&&) = delete;
```

Comment:
I want to elaborate on @CaseyCarter's comments here in case there's some ambiguity. Removing this poison pill is not editorial and also a bad idea. Consider the following type:

```cpp
namespace N {
    struct my_vector { /* ... */ };
    int const* begin(my_vector const&);
    int const* end(my_vector const&);
}
```

It's the poison pills that prevent `N::my_vector` from satisfying _`forwarding-range`_, removing them would open up some unknowable number of containers to accidentally opt-in to this concept when they definitely should not.